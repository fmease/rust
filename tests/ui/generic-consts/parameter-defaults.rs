#![feature(generic_consts)]
#![allow(incomplete_features)]

// Check that we emit a *hard* error for generic parameter defaults on free const items since we are
// not limited by backward compatibility.
// FIXME(default_type_parameter_fallback): Consider reallowing them once they work properly.

const NONE<T = ()>: Option<T> = None::<T>; //~ ERROR defaults for type parameters are only allowed

fn main() {
    let _ = NONE;
}
