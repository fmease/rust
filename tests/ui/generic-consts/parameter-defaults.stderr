error: defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
  --> $DIR/parameter-defaults.rs:8:12
   |
LL | const NONE<T = ()>: Option<T> = None::<T>;
   |            ^^^^^^

error: aborting due to previous error

